### PR TITLE
refactor: use slashed paths

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
changes once https://github.com/gakonst/ethers-rs/pull/1540 lands

## Solution
* update paths

### CODE REVIEW

**Comment by mattsse** [comment]
it should because we now have consitent `/` paths when we provide everything to solc, so solc does not introduce duplicates in the output. 

one other thing that's driving me insane is that 

> thread 'multi_runner::tests::test_fuzz' panicked at 'Test testFailFuzz(uint8) did not fail as expected.

randomly fails in cross tests https://github.com/foundry-rs/foundry/runs/7611915668?check_suite_focus=true#step:9:236

no idea why, gave up on finding a reason for this...

File: cli/src/cmd/forge/create.rs [hunk 1]
```diff
@@ -95,9 +96,11 @@ impl CreateArgs {
             compile::compile(&project, false, false)
         }?;
 
+        // output.slash_paths();
```
Comment (gakonst) [inline cli/src/cmd/forge/create.rs:99]: should this not be commented?

Comment (mattsse) [reply cli/src/cmd/forge/create.rs]: forgot to delete, no longer required as output is now slashed by default

File: cli/src/cmd/forge/inspect.rs [hunk 2]
```diff
@@ -64,8 +68,8 @@ impl Cmd for InspectArgs {
         // Build the project
         let project = modified_build_args.project()?;
         let outcome = if let Some(ref mut contract_path) = contract.path {
-            let target_path = dunce::canonicalize(&*contract_path)?;
-            *contract_path = target_path.to_string_lossy().into_owned();
+            let target_path = canonicalize(&*contract_path)?;
```
Comment (onbjerg) [inline cli/src/cmd/forge/inspect.rs:71]: Do we still need to canonicalize here?

Comment (mattsse) [reply cli/src/cmd/forge/inspect.rs:71]: yes because all paths are still absolute

Comment (mattsse) [reply cli/src/cmd/forge/inspect.rs:71]: but path slashing is now baked into `canonicolize` on windows

**Review by onbjerg — APPROVED** [review]

**Pull request merged**